src/Chapter4.hs

```diff
@@ -629,7 +647,7 @@ Can you implement a monad version of AND, polymorphic over any monad?
 🕯 HINT: Use "(>>=)", "pure" and anonymous function
 -}
 andM :: (Monad m) => m Bool -> m Bool -> m Bool
-andM = error "andM: Not implemented!"
+andM ma mb = ma >>= \a -> if a then mb else pure False
```

Comment:
👏🏼 This is absolutely correct and lazy implementation! Similar to how `&&` works itself! Great job 😊